x-pack/filebeat/input/httpjson/config_test.go

```diff
@@ -0,0 +1,418 @@
+	}
+}
+
+func TestConfigValidationCase9(t *testing.T) {
+	const expectedErr = "invalid configuration: oauth2 and api_key or authentication_scheme cannot be set simultaneously accessing config"
+	m := map[string]interface{}{
+		"authentication_scheme": "an_api_key",
+		"oauth2": map[string]interface{}{
+			"token_url": "localhost",
+			"client": map[string]interface{}{
+				"id":     "a_client_id",
+				"secret": "a_client_secret",
+			},
+		},
+		"url": "localhost",
+	}
+	cfg := common.MustNewConfigFrom(m)
+	conf := defaultConfig()
+	if err := cfg.Unpack(&conf); err.Error() != expectedErr {
+		t.Fatalf("Configuration validation failed. expecting %q error but got %q", expectedErr, err)
+	}
+}
+
+func TestConfigValidationCase10(t *testing.T) {
+	const expectedErr = "invalid configuration: both token_url and client credentials must be provided accessing 'oauth2'"
+	m := map[string]interface{}{
+		"oauth2": map[string]interface{}{},
+		"url":    "localhost",
+	}
+	cfg := common.MustNewConfigFrom(m)
+	conf := defaultConfig()
+	if err := cfg.Unpack(&conf); err.Error() != expectedErr {
+		t.Fatalf("Configuration validation failed. expecting %q error but got %q", expectedErr, err)
+	}
+}
+
+func TestConfigValidationCase11(t *testing.T) {
+	const expectedErr = "invalid configuration: unknown provider \"unknown\" accessing 'oauth2'"
+	m := map[string]interface{}{
+		"oauth2": map[string]interface{}{
+			"provider": "unknown",
+		},
+		"url": "localhost",
+	}
+	cfg := common.MustNewConfigFrom(m)
+	conf := defaultConfig()
+	if err := cfg.Unpack(&conf); err == nil || err.Error() != expectedErr {
+		t.Fatalf("Configuration validation failed. expecting %q error but got %q", expectedErr, err)
+	}
+}
+
+func TestConfigValidationCase12(t *testing.T) {
+	const expectedErr = "invalid configuration: at least one of token_url or tenant_id must be provided accessing 'oauth2'"
+	m := map[string]interface{}{
+		"oauth2": map[string]interface{}{
+			"provider": "azure",
+		},
+		"url": "localhost",
+	}
+	cfg := common.MustNewConfigFrom(m)
+	conf := defaultConfig()
+	if err := cfg.Unpack(&conf); err == nil || err.Error() != expectedErr {
+		t.Fatalf("Configuration validation failed. expecting %q error but got %q", expectedErr, err)
+	}
+}
+
+func TestConfigValidationCase13(t *testing.T) {
+	const expectedErr = "invalid configuration: only one of token_url and tenant_id can be used accessing 'oauth2'"
+	m := map[string]interface{}{
+		"oauth2": map[string]interface{}{
+			"provider":        "azure",
+			"azure.tenant_id": "a_tenant_id",
+			"token_url":       "localhost",
+		},
+		"url": "localhost",
+	}
+	cfg := common.MustNewConfigFrom(m)
+	conf := defaultConfig()
+	if err := cfg.Unpack(&conf); err == nil || err.Error() != expectedErr {
+		t.Fatalf("Configuration validation failed. expecting %q error but got %q", expectedErr, err)
+	}
+}
+
+func TestConfigValidationCase14(t *testing.T) {
```

Comment:
I'm really happy to have all the new test cases. I'm not keen on the numerical naming pattern that got established here. Maybe break for this pattern for the new tests. I'd prefer something that describes the requirements/assertion being made about the code (e.g. `TestConfigValidateRequiresCreds` or use `TestConfigValidate` with `t.Run("validate requires credentials", func(...)`).